tests/v2/validation/charts/backup_restore/backup_restore_test.go

```diff
@@ -0,0 +1,132 @@
+
+	config.LoadConfig(charts.BackupRestoreConfigurationFileKey, backupRestoreConfig)
+
+	project, err := projects.GetProjectByName(b.client, localCluster, systemProject)
+	require.NoError(b.T(), err)
+
+	logrus.Info("Checking if the backup chart is already installed...")
+	initialBackupChart, err := charts.GetChartStatus(b.client, project.ClusterID, backupChartNamespace, backupChartName)
+	require.NoError(b.T(), err)
+
+	if !initialBackupChart.IsAlreadyInstalled {
+		installBroChart(b.client)
+	}
+
+	client, err = b.client.ReLogin()
+	require.NoError(b.T(), err)
+
+	logrus.Info("Creating two users, projects, and role templates...")
+	userList, projList, roleList, err := createRancherResources(client, 2, project.ClusterID)
+	require.NoError(b.T(), err)
+
+	logrus.Info("Creating a backup of the local cluster...")
+	firstBackup, err := createBackup(client, backupName)
+	require.NoError(b.T(), err)
+	charts.VerifyBackupCompleted(client, backupSteveType, firstBackup)
+
+	backupObj, err := client.Steve.SteveType(backupSteveType).ByID(backupName)
+	require.NoError(b.T(), err)
+	backupK8Obj := &bv1.Backup{}
+	error := v1.ConvertToK8sType(backupObj.JSONResp, backupK8Obj)
+	require.NoError(b.T(), error)
+	backupFileName := backupK8Obj.Status.Filename
+
+	logrus.Info("Validating backup file is in AWS S3...")
+	backupPresent, err := validateAWSS3BackupExists(backupRestoreConfig.S3BucketName, backupRestoreConfig.S3FolderName, backupFileName)
+	require.NoError(b.T(), err)
+	assert.True(b.T(), backupPresent)
+
+	userListPostBackup, projListPostBackup, roleListPostBackup, err := createRancherResources(client, 2, project.ClusterID)
+	require.NoError(b.T(), err)
+
+	logrus.Infof("Creating a restore using backup file: %v", backupFileName)
+	completedRestore, err := createRestore(client, backupFileName)
+	require.NoError(b.T(), err)
+	restoreObj, err := client.Steve.SteveType(restoreSteveType).ByID(completedRestore.ID)
+	require.NoError(b.T(), err)
+	charts.VerifyRestoreCompleted(client, restoreSteveType, restoreObj)
+
+	logrus.Info("Restore complete, validating Rancher resources...")
+	err = verifyRancherResources(client, userList, projList, roleList)
+	require.NoError(b.T(), err)
+
+	err = verifyRancherResources(client, userListPostBackup, projListPostBackup, roleListPostBackup)
+	assert.Error(b.T(), err)
+
+	logrus.Info("Validating downstream clusters are in an Active status...")
+	clusterList = append(clusterList, backupRestoreConfig.Rke1ClusterName)
+	clusterList = append(clusterList, backupRestoreConfig.Rke2ClusterName)
+
+	for i := 0; i < len(clusterList); i++ {
+		clusterID, err := clusters.GetClusterIDByName(client, clusterList[i])
+		require.NoError(b.T(), err)
+
+		// Note: The function name mentions upgrade, but it's waiting for the cluster to become active
+		// and it logs the cluster status while waiting, therefore it's still useful for post-restore checks
+		err = clusters.WaitClusterUntilUpgrade(client, clusterID)
+		require.NoError(b.T(), err)
+	}
```

Comment:
in the readme you require an rke1 and rke2 cluster. Besides making sure they upgrade, are there any other validations that need to be done? 

If this is all that's needed, I would also suggest running the verify.go on each cluster after the upgrade.